SingularityUI/app/components/machines/usage/SlaveAggregates.jsx

```diff
@@ -0,0 +1,100 @@
+import React, { PropTypes } from 'react';
+import CircularProgressbar from 'react-circular-progressbar';
+import Utils from '../../../utils';
+import { STAT_NAMES, HUNDREDTHS_PLACE } from '../Constants';
+
+const getPctSlaveUsage = (slaves, slaveUsages, usageCallback, resourceCallback) => {
+  const totalUsage = slaveUsages.map(usageCallback)
+                                .reduce((acc, val) => acc + parseFloat(val), 0);
+
+  const totalResource = slaves.map(resourceCallback)
+                              .reduce((acc, val) => acc + parseFloat(val), 0);
+
+  return Utils.roundTo((totalUsage / totalResource) * 100, HUNDREDTHS_PLACE);
```

Comment:
This line might be good to abstract out into a helper function since it does one very specific thing.